VictoriaMetrics may try to use enterprise LTS releases

### ✅ Have you read and understood the above guidelines?

yes

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

VictoriaMetrics

### 📂 What was the exact command used to execute the script?

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/victoriametrics.sh)"

### ⚙️ What settings are you using?

- [x] Default Settings
- [x] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 13

### 📈 Which Proxmox version are you on?

pve-manager/9.1.6/71482d1833ded40a (running kernel: 6.17.13-2-pve)

### 📝 Provide a clear and concise description of the issue.

When installing VictoriaMetrics, there is a chance that the latest release on their GitHub is an LTS, enterprise release. This is not the highest version number, just the latest release.

In my case the latest release is [v1.138.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.138.0), but on the same day an LTS release was published: [v1.122.17](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.122.17).

The LTS release is the one returned by the GitHub API, grep can't find a suitable download link, and the script fails with the following error.

### 🔄 Steps to reproduce the issue.

Try to install VictoriaMetrics when the latest release on GitHub is en enterprise release.

### ❌ Paste the full error output (if available).

```
  --- Last 20 lines of log ---
Setting up libnss-systemd:amd64 (257.9-1~deb13u1) ...
Setting up e2fsprogs (1.47.2-3+b10) ...
Setting up wtmpdb (0.73.0-3+deb13u1) ...
Setting up python3-requests (2.32.3+dfsg-5+deb13u1) ...
Setting up libpam-systemd:amd64 (257.9-1~deb13u1) ...
Setting up bind9-host (1:9.20.21-1~deb13u1) ...
Setting up ssh (1:10.0p1-7+deb13u1) ...
Setting up bind9-dnsutils (1:9.20.21-1~deb13u1) ...
Processing triggers for man-db (2.13.1-1) ...
Processing triggers for dbus (1.16.2-2) ...
Processing triggers for shared-mime-info (2.4-5+b2) ...
Processing triggers for procps (2:4.0.4-9) ...
Processing triggers for debianutils (5.23.2) ...
Processing triggers for libc-bin (2.41-12+deb13u2) ...
Processing triggers for postfix (3.10.5-1~deb13u1) ...
Restarting postfix
[2026-03-28 17:54:51] [OK] Updated Container OS
[2026-03-28 17:54:52] [INFO] Getting latest version of VictoriaMetrics
[2026-03-28 17:54:52] [ERROR] in line 19: exit code 1 (General error / Operation not permitted): while executing command grep -E '^victoria-metrics-linux-amd64-v[0-9.]+\.tar\.gz$'
[2026-03-28 17:54:52] [ERROR] in line 19: exit code 1 (General error / Operation not permitted): while executing command victoriametrics_filename=$(curl -fsSL "https://api.github.com/repos/VictoriaMetrics/VictoriaMetrics/releases/latest" | jq -r '.assets[].name' | grep -E '^victoria-metrics-linux-amd64-v[0-9.]+\.tar\.gz$')
  -----------------------------------
```

[victoriametrics-104-e4ddea2a.log](https://github.com/user-attachments/files/26324849/victoriametrics-104-e4ddea2a.log)

### 🖼️ Additional context (optional).

The GitHub API response that is piped through jq and then filtered by grep is as follows:

[github-api-response.json](https://github.com/user-attachments/files/26324850/github-api-response.json)

--- a/ct/victoriametrics.sh
+++ b/ct/victoriametrics.sh
@@ -34,24 +34,26 @@ function update_script() {
     [[ -f /etc/systemd/system/victoriametrics-logs.service ]] && systemctl stop victoriametrics-logs
     msg_ok "Stopped Service"
 
-    victoriametrics_filename=$(curl -fsSL "https://api.github.com/repos/VictoriaMetrics/VictoriaMetrics/releases/latest" |
-      jq -r '.assets[].name' |
-      grep -E '^victoria-metrics-linux-amd64-v[0-9.]+\.tar\.gz$')
-    vmutils_filename=$(curl -fsSL "https://api.github.com/repos/VictoriaMetrics/VictoriaMetrics/releases/latest" |
-      jq -r '.assets[].name' |
-      grep -E '^vmutils-linux-amd64-v[0-9.]+\.tar\.gz$')
+    victoriametrics_release=$(curl -fsSL "https://api.github.com/repos/VictoriaMetrics/VictoriaMetrics/releases" |
+      jq -r '.[] | select(.assets[].name | match("^victoria-metrics-linux-amd64-v[0-9.]+.tar.gz$")) | .tag_name' |
+      head -n 1)
 
-    fetch_and_deploy_gh_release "victoriametrics" "VictoriaMetrics/VictoriaMetrics" "prebuild" "latest" "/opt/victoriametrics" "$victoriametrics_filename"
-    fetch_and_deploy_gh_release "vmutils" "VictoriaMetrics/VictoriaMetrics" "prebuild" "latest" "/opt/victoriametrics" "$vmutils_filename"
+    msg_debug "Using release $victoriametrics_release"
+
+    victoriametrics_filename="victoria-metrics-linux-amd64-${victoriametrics_release}.tar.gz"
+    vmutils_filename="vmutils-linux-amd64-${victoriametrics_release}.tar.gz"
+
+    fetch_and_deploy_gh_release "victoriametrics" "VictoriaMetrics/VictoriaMetrics" "prebuild" "$victoriametrics_release" "/opt/victoriametrics" "$victoriametrics_filename"
+    fetch_and_deploy_gh_release "vmutils" "VictoriaMetrics/VictoriaMetrics" "prebuild" "$victoriametrics_release" "/opt/victoriametrics" "$vmutils_filename"
 
     if [[ -f /etc/systemd/system/victoriametrics-logs.service ]]; then
       vmlogs_filename=$(curl -fsSL "https://api.github.com/repos/VictoriaMetrics/VictoriaLogs/releases/latest" |
         jq -r '.assets[].name' |
-        grep -E '^victoria-logs-linux-amd64-v[0-9.]+\.tar\.gz$')  
+        grep -E '^victoria-logs-linux-amd64-v[0-9.]+\.tar\.gz$')
       vlutils_filename=$(curl -fsSL "https://api.github.com/repos/VictoriaMetrics/VictoriaLogs/releases/latest" |
         jq -r '.assets[].name' |
         grep -E '^vlutils-linux-amd64-v[0-9.]+\.tar\.gz$')
-        
+
       fetch_and_deploy_gh_release "victorialogs" "VictoriaMetrics/VictoriaLogs" "prebuild" "latest" "/opt/victoriametrics" "$vmlogs_filename"
       fetch_and_deploy_gh_release "vlutils" "VictoriaMetrics/VictoriaLogs" "prebuild" "latest" "/opt/victoriametrics" "$vlutils_filename"
     fi

--- a/install/victoriametrics-install.sh
+++ b/install/victoriametrics-install.sh
@@ -14,16 +14,16 @@ network_check
 update_os
 
 msg_info "Getting latest version of VictoriaMetrics"
-victoriametrics_filename=$(curl -fsSL "https://api.github.com/repos/VictoriaMetrics/VictoriaMetrics/releases/latest" |
-  jq -r '.assets[].name' |
-  grep -E '^victoria-metrics-linux-amd64-v[0-9.]+\.tar\.gz$')
-vmutils_filename=$(curl -fsSL "https://api.github.com/repos/VictoriaMetrics/VictoriaMetrics/releases/latest" |
-  jq -r '.assets[].name' |
-  grep -E '^vmutils-linux-amd64-v[0-9.]+\.tar\.gz$')
-msg_ok "Got latest version of VictoriaMetrics"
 
-fetch_and_deploy_gh_release "victoriametrics" "VictoriaMetrics/VictoriaMetrics" "prebuild" "latest" "/opt/victoriametrics" "$victoriametrics_filename"
-fetch_and_deploy_gh_release "vmutils" "VictoriaMetrics/VictoriaMetrics" "prebuild" "latest" "/opt/victoriametrics" "$vmutils_filename"
+victoriametrics_release=$(curl -fsSL "https://api.github.com/repos/VictoriaMetrics/VictoriaMetrics/releases" |
+  jq -r '.[] | select(.assets[].name | match("^victoria-metrics-linux-amd64-v[0-9.]+.tar.gz$")) | .tag_name' |
+  head -n 1)
+victoriametrics_filename="victoria-metrics-linux-amd64-${victoriametrics_release}.tar.gz"
+vmutils_filename="vmutils-linux-amd64-${victoriametrics_release}.tar.gz"
+msg_ok "Got version $victoriametrics_release of VictoriaMetrics"
+
+fetch_and_deploy_gh_release "victoriametrics" "VictoriaMetrics/VictoriaMetrics" "prebuild" "$victoriametrics_release" "/opt/victoriametrics" "$victoriametrics_filename"
+fetch_and_deploy_gh_release "vmutils" "VictoriaMetrics/VictoriaMetrics" "prebuild" "$victoriametrics_release" "/opt/victoriametrics" "$vmutils_filename"
 
 read -r -p "${TAB3}Would you like to add VictoriaLogs? <y/N> " prompt
 

